app/notify_client/job_api_client.py

```diff
@@ -112,6 +121,15 @@ def create_job(self, job_id, service_id, scheduled_for=None):
             scheduled_for = JobApiClient.convert_user_time_to_utc(scheduled_for)
             data.update({"scheduled_for": scheduled_for})
 
+        if template_id:
+            data.update({"template_id": template_id})
+        if original_file_name:
+            data.update({"original_file_name": original_file_name})
+        if notification_count:
+            data.update({"notification_count": notification_count})
+        if valid:
+            data.update({"valid": valid})
```

Comment:
I realize this is written this way above, but it seems overly verbose and for no real performance gain or anything.  We could make adding to the existing `data` dictionary a little clearer:

```suggestion
        if template_id:
            data["template_id"] = template_id
        if original_file_name:
            data["original_file_name"] = original_file_name
        if notification_count:
            data["notification_count"] = notification_count
        if valid:
            data["valid"] = valid
```

This will have the same effect of either creating and setting a new key, or overwriting an existing one. :-)